lib/graphql/language/parser.y

```diff
@@ -293,6 +297,47 @@ rule
     | enum_type_definition
     | input_object_type_definition
 
+  type_system_extension:
+      schema_extension
+    | type_extension
+
+  schema_extension:
+      EXTEND SCHEMA directives_list_opt LCURLY operation_type_definition_list RCURLY { return make_node(:SchemaExtension, position_source: val[0], directives: val[2], **val[4]) }
+    | EXTEND SCHEMA directives_list { return make_node(:SchemaExtension, position_source: val[0], directives: val[2]) }
+
+  type_extension:
+      scalar_type_extension
+    | object_type_extension
+    | interface_type_extension
+    | union_type_extension
+    | enum_type_extension
+    | input_object_type_extension
+
+  scalar_type_extension: EXTEND SCALAR name directives_list { return make_node(:ScalarTypeExtension, name: val[2], directives: val[3], position_source: val[0]) }
+
+  object_type_extension:
+      /* TODO - This first one shouldn't be necessary but parser is getting confused */
+      EXTEND TYPE name implements LCURLY field_definition_list RCURLY { return make_node(:ObjectTypeExtension, name: val[2], interfaces: val[3], directives: [], fields: val[5], position_source: val[0]) }
```

Comment:
Something fishy is going on in the parser. I tried debugging (without success) why this one is required. 🤔 

You would think the 2nd line covers this case, but without it, this:

```
extend type Something implements A { field: String }
```

Gets parsed as:

`ObjectTypeExtension` + `OperationDefinition`

Instead of:

`ObjectTypeExtension`